runtime/logging.go

```diff
@@ -127,6 +137,13 @@ func (h *LoggingHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				// metrics endpoint does so when the client accepts it (e.g. prometheus)
 				fields["resp_body"] = "[compressed payload]"
 
+			case gzipAccepted(r.Header) && gzipReceived(w.Header()) && (isDataEndpoint(r) || isCompileEndpoint(r)):
+				// data and compile endpoints might compress the response
+				reader := bytes.NewReader(recorder.buf.Bytes())
+				gzReader, _ := gzip.NewReader(reader)
+				plainOutput, _ := io.ReadAll(gzReader)
```

Comment:
Are we really certain that those ignored are never happening?

Reply:
I've added checks for error, please check.